lib/services/northBound/contextServer.js

```diff
@@ -862,12 +862,18 @@ function handleQueryNgsi2(req, res, next) {
 
     logger.debug(context, 'Handling query from [%s]', req.get('host'));
     var contextEntity = {};
-    contextEntity.id = req.query.id;
-    contextEntity.type = req.query.type;
-    var queryAtts = [];
-    if(req.query.attrs) {
-        queryAtts = req.query.attrs.split(',');
+
+    // Heuristic: we are assuming that the entities array in POST /v2/op/query body has only one item
```

Comment:
What happens now if entities array in the body has more than one entity? The code will just return the result for the first one... This may be a bit confusing for the user. We could check if there is more than one entity and return an error in that case.

Reply:
This is similar to https://github.com/dcalvoalonso/iotagent-node-lib/pull/8#discussion_r320635876. We can leave this case out of CB-IOTA integrations.

Reply:
Fixed in 5ea1b772657033d0e4691b1b8538e17de5113749